argoproj/argo
# feat(controller): Use S3 for node offload + workflow archiving. POC

Signed-off-by: Alex Collins <alex_collins@intuit.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).



# Testing Instructions

You need to run `argoproj/workflow-controller:off-poc` with additional configuration:


```yaml
      # Storage: s3, mysql, or postgres maybe configured.
      #
      # If mulitple are configured then the following rules are used:
      #
      # 1. For node status offloading, S3 is most preference, for archiving, least prefered.
      # 2. MySQL is prefered to PostgreSQL
      #
      # S3 is most prefered (and therefore implicitly recommeded) as it can be expected to be faster,
      # and relies less on the querying capability of an SQL database.
      #
      # S3 is least prefered for workflow archiving. While is is easier to set-up that an SQL database, S3 does not
      # support (out of the box) indexed backed (i.e. fast) querying capabilities for listing archived workflows.
      # List operations will be slow when any significant number of workflows are archived.
      # Unlike MySQL or Postgres, S3 enforces TTLs by setting them on the bucket.
      # You MUST configure different buckets for the archive and offloads if you wish to TTL archived workflows.
      s3: |
        nodeStatusOffload: # ... as artifactRepository.s3
        archive: # ... as artifactRepository.s3
```

See #4560
